From 24779d90ad56f6e35b992ceef095c71f4e5ab3f7 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Mon, 4 Dec 2017 10:20:34 +0800
Subject: [PATCH 2968/5242] MLK-15985-3: ARM: dts: imx6sx: set frequency for
 S20_3LE format

commit  2968714df1fae03882770cca7b4ea380f43e9be5 from
https://source.codeaurora.org/external/imx/linux-imx.git

Select a proper frequency for S20_3LE/S24_LE/S16_LE. and disable
the SSI2 when mqs enabled, for mqs only support 24.576MHz mclk.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6sx-sdb-mqs.dts |    9 +++++++++
 arch/arm/boot/dts/imx6sx-sdb.dtsi    |    7 +++++++
 2 files changed, 16 insertions(+)

diff --git a/arch/arm/boot/dts/imx6sx-sdb-mqs.dts b/arch/arm/boot/dts/imx6sx-sdb-mqs.dts
index e5ff9fb..bf388ac 100644
--- a/arch/arm/boot/dts/imx6sx-sdb-mqs.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb-mqs.dts
@@ -11,6 +11,11 @@
 
 #include "imx6sx-sdb.dts"
 / {
+
+	sound {
+		status = "disabled";
+	};
+
 	sound-mqs {
 		compatible = "fsl,imx6sx-sdb-mqs",
 				"fsl,imx-audio-mqs";
@@ -39,6 +44,10 @@
 	status = "okay";
 };
 
+&ssi2 {
+	status = "disabled";
+};
+
 &sdma {
 	gpr = <&gpr>;
 	/* SDMA event remap for SAI1 */
diff --git a/arch/arm/boot/dts/imx6sx-sdb.dtsi b/arch/arm/boot/dts/imx6sx-sdb.dtsi
index 6166bbf..752953b 100644
--- a/arch/arm/boot/dts/imx6sx-sdb.dtsi
+++ b/arch/arm/boot/dts/imx6sx-sdb.dtsi
@@ -634,6 +634,13 @@
 };
 
 &ssi2 {
+	assigned-clocks = <&clks IMX6SX_CLK_PLL4>,
+			  <&clks IMX6SX_PLL4_BYPASS>,
+			  <&clks IMX6SX_CLK_SSI2_SEL>;
+	assigned-clock-parents = <&clks IMX6SX_CLK_OSC>,
+				 <&clks IMX6SX_CLK_PLL4>,
+				 <&clks IMX6SX_CLK_PLL4_AUDIO_DIV>;
+	assigned-clock-rates = <737280000>, <0>, <0>;
 	status = "okay";
 };
 
-- 
1.7.9.5

